compiler/rustc_target/src/abi/call/s390x.rs

```diff
@@ -1,19 +1,46 @@
-// FIXME: The assumes we're using the non-vector ABI, i.e., compiling
-// for a pre-z13 machine or using -mno-vx.
+use crate::abi::call::{ArgAbi, FnAbi, Reg, RegKind};
+use crate::abi::{Abi, HasDataLayout, Size, TyAbiInterface, TyAndLayout};
+use crate::spec::{HasS390xVector, HasTargetSpec};
 
-use crate::abi::call::{ArgAbi, FnAbi, Reg};
-use crate::abi::{HasDataLayout, TyAbiInterface};
-use crate::spec::HasTargetSpec;
+#[derive(Debug, Clone, Copy, PartialEq)]
+enum ABI {
+    NoVector, // no-vector ABI, i.e., compiling for a pre-z13 machine or using -C target-feature=-vector
+    Vector, // vector ABI, i.e., compiling for a z13 or later machine or using -C target-feature=+vector
+}
```

Comment:
Once ABI check for s390x vector type (by extending https://github.com/rust-lang/rust/pull/127731 's ABI check) is implemented, the complexity of discriminating between ABIs should be unnecessary, since we can always assume a vector ABI for a vector type.

(It also fixes the incorrect handling of `#[target_feature(enable)]` cases.)